### PR TITLE
add stubs + some optimizations

### PR DESCRIPTION
**Pull request by rewweRrr**

### CODE REVIEW

File: src/common/api.service.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+export abstract class ApiService {
+  protected debug: boolean = window.localStorage.getItem("debug") === "1";
+
+  private readonly fileUrl: string = "/assets/json/";
+
+  constructor(private http: Http) {
+  }
+
+  protected getDataFromFile<T>(fileName: string): Observable<T> {
+    return this.http.get(this.fileUrl + fileName).map((res) => res.json() || {})
```
Comment (defesteban) [inline src/common/api.service.ts]: res -> result, response?
+ add type for 'response' variable

File: src/sms-parser/modules/sms/components/sms-list.component.ts [hunk 2]
```diff
@@ -1,42 +1,54 @@
...
+    this.filteredListSubscription = this.smsService.getFilteredMessages({maxCount: this._maxCount}).subscribe((smsList: SmsModel[]) => {
       this._allSmsList = smsList;
       setTimeout(() => {
         PageUtils.scrollTop(this.scrollTop);
       });
     });
   }
+
+  //optimization
+  _trackByFn(index: number, item: SmsModel) {
```
Comment (defesteban) [inline src/sms-parser/modules/sms/components/sms-list.component.ts]: Add comment why we need to use 'index'.
Add return statement.

File: src/sms-parser/modules/sms/sms.service.ts [hunk 3]
```diff
@@ -2,36 +2,56 @@ import {Injectable} from "@angular/core";
...
-      });
+  getFilteredMessages(filter: SmsFilterModel): Observable<SmsModel[]> {
+    return Observable.create((subscriber: Subscriber<SmsModel[]>) => {
+      let sms = window["SMS"];
+
+      let listSms: SmsModel[] = [];
+
+      if (this.debug) {
+        this.getDataFromFile<SmsVirtualModel[]>("sms-list.json").subscribe((data: SmsVirtualModel[]) => {
+          listSms = this.convertSmsVirtualModel(data);
```
Comment (defesteban) [inline src/sms-parser/modules/sms/sms.service.ts]: Code is duplicated.
Create callback.

File: src/common/api.service.ts (hunk 1 above)
Comment (rewweRrr) [reply src/common/api.service.ts]: done

File: src/sms-parser/modules/sms/components/sms-list.component.ts (hunk 2 above)
Comment (rewweRrr) [reply src/sms-parser/modules/sms/components/sms-list.component.ts]: done

File: src/sms-parser/modules/sms/sms.service.ts (hunk 3 above)
Comment (rewweRrr) [reply src/sms-parser/modules/sms/sms.service.ts]: done

**Review by defesteban — APPROVED** [review]

**Pull request merged**